docs/Project-Configuration/Project-Configuration-Files.md

```diff
@@ -464,22 +464,27 @@ timeout:
         python buildscripts/hang_analyzer.py
 ```
 
+**Exec timeout: exec_timeout_secs**
 You can customize the points at which the "timeout" conditions are
 triggered. To cause a task to stop (and fail) if it doesn't complete
 within an allotted time, set the key `exec_timeout_secs` on the project
 or task to the maximum allowed length of execution time. This timeout
 defaults to 6 hours. `exec_timeout_secs` can only be set on the project
 or on a task. It cannot be set on functions.
 
-You can also set exec_timeout_secs using [timeout.update](Project-Commands.md#timeoutupdate). 
+You can also set exec_timeout_secs using [timeout.update](Project-Commands.md#timeoutupdate).
 
+**Idle timeout: timeout_secs**
 You may also force a specific command to trigger a failure if it does
 not appear to generate any output on `stdout`/`stderr` for more than a
 certain threshold, using the `timeout_secs` setting on the command. As
 long as the command does not appear to be idle it will be allowed to
 continue, but if it does not write any output for longer than
-`timeout_secs` then the timeout handler will be triggered. This timeout
-defaults to 2 hours.
+`timeout_secs` then the timeout handler will be triggered. If timeout_secs
+is specified for a command that runs in a pre or post block, it will
+only be set in pre or post if pre_error_fails_task or post_error_fails_task
```

Comment:
Since it's set at the command level, it seems like it would be more consistent/desirable behavior to have the idle timeout always be respected as the timeout, regardless of whether `pre/post_error_fails_task` is set. Whether or not pre/post is relevant to the end task status is an independent concern from how/when the command times out.

Reply:
I changed the code above to set tc.setCurrentIdleTimeout without any condition and removed that line in the docs.